src/pages/developer-tools/ide-integrated-development-environments/index.md

```diff
@@ -3,13 +3,19 @@ title: IDE Integrated Development Environments
 ---
 ## IDE Integrated Development Environments
 
-This is a stub. <a href='https://github.com/freecodecamp/guides/tree/master/src/pages/developer-tools/ide-integrated-development-environments/index.md' target='_blank' rel='nofollow'>Help our community expand it</a>.
+**IDE** or Integrated Development Environment is a type of computer program which is used to make software. Usually all the components needed for software development are contained in this one program, allowing for faster development.  
 
-<a href='https://github.com/freecodecamp/guides/blob/master/README.md' target='_blank' rel='nofollow'>This quick style guide will help ensure your pull request gets accepted</a>.
+In general, an **IDE** may contain a compiler, interpreter, or both.  However, some IDE’s do not contain either of those components. Most **IDEs** tend to have built in version control systems such as Git which is accessed using a GUI (graphic user interface). 
+Many modern **IDE's** also have a class browser, an object browser, and a class hierarchy diagram, for use in object-oriented software development.
 
-<!-- The article goes here, in GitHub-flavored Markdown. Feel free to add YouTube videos, images, and CodePen/JSBin embeds  -->
+Here are some examples of **IDE** programs which are free and most of which are open source. The languages they are used in are listed to next to them. :smiley: Shown below in no particular order :sparkles:
+1. [Eclipse IDE](https://www.eclipse.org/ide/) - Java, C/C++, PHP, and Web Development (HTML, CSS, JavaScript, Java ME)
+2. [IntelliJ IDEA](https://www.jetbrains.com/idea/) - Java and Kotlin. The same platform is available in _many_ languages, for a yearly fee. 
+3. [NetBeans](https://netbeans.org/features/index.html) - Java, C/C++, Groovy, PHP, HTML5 Web Development (HTML, CSS, JavaScript, NodeJS)
+4. [Microsoft Visual Studio Community 2017](https://www.visualstudio.com/vs/community/) - Code in C#, Visual Basic, F#, C++, HTML, JavaScript, TypeScript, Python, and more. 
+5. [Android Studio](https://developer.android.com/studio/index.html) - Java, Kotlin, C++, Groovy
 
 #### More Information:
-<!-- Please add any articles you think might be helpful to read before writing the article -->
-
+Many developer will also use a text editor such as [Visual Studio Code](https://code.visualstudio.com/), [Atom Editor](https://www.atom.io), or [Sublime Text Editor](https://www.sublimetext.com/) and use 3rd party extensions to add the features which they would get from a traditional **IDE**. **IDEs** generally use more of a computers resources than a text editor.
 
+Written by [PeterCr](https://www.github.com/petercr)
```

Comment:
Thanks a lot for contributing @petercr! 

We prefer not to put names on articles.
Ideally, the Guide will constantly be evolving and improving. Anybody can jump in to make something even better.
Putting a name on an article (or the equivalent in code, a comment that says who created the file) might discourage others from contributing to it, as it defines "ownership." People don't want to touch something that's the "property" of someone else, even if they think they could help make it better. 
It's also hard to define when it's no longer valid (when are there enough changes to say that it's no longer written just by you).

Your authorship will be tracked here on GitHub, so you can always prove that you wrote this. For the sake of making sure the Guide keeps getting better over time, it's best not to add names to the content.

I don't have the time to do a full review of this, but I believe @texas2010 was planning on looking at this.

Reply:
@systimotic Thank you very much for letting me know. I _wasn't sure_ if the first pull request had worked, so I just added in the author part so I could make sure the pull request went through. But thanks again for letting me know 😃 :+1: